src/services/email.service.js

```
@@ -0,0 +1,95 @@
+
+  oauth2Client.setCredentials({
+    refresh_token: process.env.REFRESH_TOKEN,
+  });
+
+  const accessTokenObj = await oauth2Client.getAccessToken();
+
+  const transporter = nodemailer.createTransport({
+    service: 'gmail',
+    auth: {
+      type: 'OAuth2',
+      user: 'v.hulaievych@gmail.com',
+      emailConfig,
+      accessToken: accessTokenObj.token,
+    },
+  });
+
+  return transporter;
+}
+
+async function send({ email, subject, html }) {
+  const transporter = await createTransporter();
+
+  return transporter.sendMail({
+    to: email,
+    subject,
+    html,
+  });
+}
+
+async function sendActivationEmail(email, activationToken) {
+  const href = `${process.env.CLIENT_HOST}/activate?token=${activationToken}`;
+  const subject = 'Activation account';
+  const html = `
+    <h1>activation</h1>
+    <a href="${href}">${href}</a>
+  `;
+
+  try {
+    await send({ email, subject, html });
+    console.log('✅ Activation email sent to', email);
+  } catch (error) {
+    console.error('Error sending email:', error.message);
+    throw ApiError.badRequest('Server problem', error);
+  }
+}
+
+async function sendResetPasswordEmail(email, resetPasswordToken) {
+  const href = `${process.env.CLIENT_HOST}/reset-password/${email}/${resetPasswordToken}`;
+  const subject = 'Reset your password';
+  const html = `
+    <h1>Reset your password</h1>
+    <a href="${href}">${href}</a>
+  `;
+
+  try {
+    await send({ email, subject, html });
+    console.log('✅ Reset password email sent to', email);
+  } catch (error) {
+    console.error('Error sending email:', error.message);
+  }
+}
+
+async function sendChangedEmail(email) {
+  const subject = 'Email changed';
+  const html = `
+    <h1>Your email has been change to ${email}</h1>
+  `;
+
+  try {
+    await send({ email, subject, html });
+    console.log('✅ Reset password email sent to', email);
+  } catch (error) {
+    console.error('Error sending email:', error.message);
+  }
```

Similar to the password reset function, this `catch` block swallows the error. If the notification email fails to send, the error should be re-thrown so the application can handle it appropriately instead of silently failing.